web/services/updateUserGoalsService.php

```diff
@@ -40,7 +40,7 @@
 
     $parser->read();
 
-    if ($parser->name == 'updateUserGoals')
+    if ($parser->name == 'userGoals')
```

Comment:
Why did you have to change this one?

Reply:
It was wrong, the top-level tag for this service is "userGoals" and for that reason it never initialized the `$sid` variable like other similar services do:
```
<?xml version="1.0" ?>
<userGoals sid='12341235123591'> <!-- sid is optional, actually never used -->
  <userGoal>
    <userId>2</userId>
    <id>2</id>
    <extraTime>11</extraTime>
    <init>2021-09-30</init>
    <end>2022-09-28</end>
  </userGoal>
</userGoals>